source/ControllableHero.hx

```diff
@@ -0,0 +1,69 @@
+package;
+
+import flixel.FlxSprite;
+import flixel.FlxG;
+
+class ControllabeHero extends FlxSprite
+{
+    static inline var SPEED:Float = 50;
+
+    public function new(X:Float = 0, Y:Float = 0) {
+        super(X, Y);
+
+        this.makeGraphic(30, 30, 0xFFFFFF00);
+    }
+    
+    override public function update(elapsed:Float):Void
+    {
+        if (FlxG.keys.pressed.UP)
+        {
+            // The up arrow key is currently pressed
+            // This code is executed every frame, while the key is pressed
+            this.velocity.y = -SPEED;
+        }
+        if (FlxG.keys.justReleased.UP)
+        {
+            // The left arrow key has just been released
+            // This code is only executed once, on the frame immediately after the key has been released
+            this.velocity.y = 0;
+        }
+        if (FlxG.keys.pressed.RIGHT)
+        {
+            // The up arrow key is currently pressed
+            // This code is executed every frame, while the key is pressed
+            this.velocity.x = SPEED;
+        }
+        if (FlxG.keys.justReleased.RIGHT)
+        {
+            // The left arrow key has just been released
+            // This code is only executed once, on the frame immediately after the key has been released
+            this.velocity.x = 0;
+        }
+        if (FlxG.keys.pressed.DOWN)
+        {
+            // The up arrow key is currently pressed
+            // This code is executed every frame, while the key is pressed
+            this.velocity.y = SPEED;
+        }
+        if (FlxG.keys.justReleased.DOWN)
+        {
+            // The left arrow key has just been released
+            // This code is only executed once, on the frame immediately after the key has been released
+            this.velocity.y = 0;
+        }
+        if (FlxG.keys.pressed.LEFT)
+        {
+            // The up arrow key is currently pressed
+            // This code is executed every frame, while the key is pressed
+            this.velocity.x = -SPEED;
+        }
+        if (FlxG.keys.justReleased.LEFT)
+        {
+            // The left arrow key has just been released
+            // This code is only executed once, on the frame immediately after the key has been released
+            this.velocity.x = 0;
+        }
```

Comment:
This works out okay, but it's definitely very cumbersome (especially considering it's in a method like `update()` that's going to be dense with logic). This also has the probably undesirable side-effect of making the inputs checked later take priority over inputs checked earlier (Left + Right = move left, for example).

Some suggestions for improvement:
 * Moving all of this to a private method would be a good way to help keep things organized.
 * I think you could get away with just having the mostly-repeated documentation once at the top of the whole conditional block,
 * In a case like this where we just have constant velocities, you could cut out all of the "set back to zero" logic by just putting it at the start of update() like this:
   ```
   velocity.x = 0
   velocity.y = 0
   if (FlxG.keys.pressed.UP) 
   {
     ...
   }
   ...
   ```

Changes aren't necessary for merging, but I wanted to bring it up for the future health of the codebase.